plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -140,11 +140,14 @@ private static void sendPostRequest(String parametersString) {
       connection.setDoOutput(true);
       connection.setRequestMethod("POST");
       connection.setRequestProperty("Content-Length", Integer.toString(parametersString.length()));
+      connection.setReadTimeout(3000);  // milliseconds
 
       try (Writer writer = new OutputStreamWriter(connection.getOutputStream(), "UTF-8")) {
         writer.write(parametersString);
         writer.flush();
       }
+
+      connection.getResponseCode();  // This call is important to ensure posting the request.
```

Comment:
Oh, I found a way to make it work without `getResponseCode()`: calling `setFixedLengthStreamingMode()`. We know the content-length beforehand, and by setting this fixed length, the HTTP connection knows exactly when to send the message.

Some info: http://stackoverflow.com/questions/26808136/why-httpurlconnection-does-not-send-data-unless-i-try-to-receive-something